### PR TITLE
#15: add formatting stage

### LINKED ISSUE #15: add formatting stage so that we keep things consistent

### PR DESCRIPTION
**Pull request by cwschilly**
Fixes #15

### CODE REVIEW

**Review by fnrizzi — CHANGES_REQUESTED** [review]
can you please put the formatting in the same pipeline as the rest? 
the formatting is the FIRST thing that is checked. Formatting should be a precondition for the CI to move forward. 

if formatting passes, then run all tests etc. If formatting fails, the CI stops immediately.

File: pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ requires-python = ">=3.8"
 dependencies = [
     'docutils', 'importlib-metadata',
     'numpy', 'scipy', 'matplotlib', 'pdoc',
-    'pytest', 'pytest-mpi>=0.6'
+    'pytest', 'pytest-mpi>=0.6', 'pylint'
```
Comment (fnrizzi) [inline pyproject.toml:12]: i dont think we need this as a requirement. This is only needed for the CI. agree?

File: .github/workflows/install-and-test.yaml [hunk 2]
```diff
@@ -49,6 +49,11 @@ jobs:
           pip list
           pip install .
 
+      - name: Check Formatting (Pylint)
+        run: |
+          pylint --recursive y --rcfile .pylintrc .
+        continue-on-error: false
```
Comment (fnrizzi) [inline .github/workflows/install-and-test.yaml:55]: nice!

File: pyproject.toml (hunk 1 above)
Comment (cwschilly) [reply pyproject.toml:12]: Yes; should I just `pip install pylint` as a step in CI?

**Comment by cwschilly** [comment]
Wait to merge until #40 is resolved

**Review by fnrizzi — APPROVED** [review]

**Pull request merged**